src/NuGet.Jobs.GitHubIndexer/RepoUtils.cs

```diff
@@ -0,0 +1,137 @@
+                var nodePath = string.IsNullOrWhiteSpace(currentPath) ?
+                    node.Path :
+                    Path.Combine(currentPath, node.Path);
+
+                if (node.TargetType == LibGit2Sharp.TreeEntryTargetType.Tree)
+                {
+                    files.AddRange(ListTree((LibGit2Sharp.Tree)repo.Lookup(node.Target.Id), nodePath, repo));
+                    continue;
+                }
+
+                // Only handle files
+                if (node.TargetType != LibGit2Sharp.TreeEntryTargetType.Blob)
+                {
+                    continue;
+                }
+
+                var blobSize = ((LibGit2Sharp.Blob)node.Target).Size;
+                var nodeInfo = new GitFileInfo(nodePath, blobSize);
+                files.Add(nodeInfo);
+            }
+
+            return files;
+        }
+
+        /// <summary>
+        /// Parses a packages.config file and retreives NuGet dependencies in it
+        /// </summary>
+        /// <param name="fileStream">Stream to the packages.config file</param>
+        /// <param name="repoName">The name of the repository containing the file (for logging purposes)</param>
+        /// <returns>A list of NuGet packages listed in this config file</returns>
+        public IReadOnlyList<string> ParsePackagesConfig(Stream fileStream, string repoName)
+        {
+            try
+            {
+                return new PackagesConfigReader(fileStream)
+                    .GetPackages()
+                    .Select(p => p.PackageIdentity.Id)
+                    .Where(IsValidPackageId)
+                    .ToList();
+            }
+            catch (Exception e)
+            {
+                _logger.LogError(0, e, "[{RepoName}] Cannot parse file", repoName);
+            }
+
+            return Array.Empty<string>();
+        }
+
+        /// <summary>
+        /// Parses any XML file containign PackageReference nodes and retreives those NuGet dependencies
+        /// </summary>
+        /// <param name="fileStream">Stream to the packages.config file</param>
+        /// <param name="repoName">The name of the repository containing the file (for logging purposes)</param>
+        /// <returns>A list of NuGet packages listed in this config file</returns>
+        public IReadOnlyList<string> ParseProjFile(Stream fileStream, string repoName)
+        {
+            try
+            {
+                var projDocument = XDocument.Load(fileStream);
+                var refs =
+                    projDocument
+                        .DescendantNodes()
+                        .OfType<XElement>()
+                        .Where(node => (node).Name.LocalName.Equals("PackageReference"));
+
+                return refs
+                    .Select(p => p.Attribute("Include"))
+                    .Where(includeAttr => includeAttr != null)// Select all that have an "Include" attribute
+                    .Where(includeAttr => !includeAttr.ToString().Contains("$"))
```

Comment:
You can remove the `ToString()` on line 109 if you do line 110 first:

```suggestion
                    .Where(includeAttrValue=> !includeAttrValue.Contains("$"))
```